hooks/useCustomStyle.tsx

```diff
@@ -0,0 +1,148 @@
+import { useEffect, useState } from "react";
+import { createPortal } from "react-dom";
+
+function isMobile() {
+  const userAgent = navigator.userAgent || navigator.vendor ;
+  return /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(userAgent);
+}
+
+// const isMobile = () => /iPhone|Android/i.test(globalThis.navigator.userAgent);
+
+function getPadding(mainElement: HTMLElement) {
+  return Number(getComputedStyle(mainElement).paddingLeft.replace(/[^0-9]/g, "")) ;
+}
+
+function getPseudoElementWidth(currentElement: HTMLElement, pseudoName : string) {
+
+    const computedStyle = getComputedStyle(currentElement, pseudoName);
+
+    // dynamically create element because of computing content width dynamically
+    const customCssElement = document.createElement('div');
+    customCssElement.style.position = 'absolute';
+    customCssElement.style.visibility = 'hidden';
+    customCssElement.style.pointerEvents = 'none';
+    customCssElement.className = 'invisible-div';
+    const stylesToCopy = [
+      'content',
+      'position',
+      'transform',
+      'display',
+      'position',
+      'width',
+      'height',
+      'border',
+      'text-align',
+      'padding',
+      'border-radius',
+      'font-family',
+      'white-space',
+      'margin',
+    ];
+
+    stylesToCopy.forEach(style => customCssElement.style[style as any] = computedStyle.getPropertyValue(style));
+
+    customCssElement.textContent = customCssElement.style.content.replace(/['"]+/g, '');
+    currentElement.appendChild(customCssElement);
+
+    const rect = customCssElement.getBoundingClientRect();
+    customCssElement.remove();
+    return rect.width;
+}
+
+function getCurrentScroll(currentElement : HTMLElement) {
+    const containerElement = currentElement.closest('.code-container')!
+    return containerElement?.scrollLeft
+}
+
+function changeTwoSlashXPos( currentElement : HTMLElement , currentScroll : number) {
+
+    const screenWidth = globalThis.window.outerWidth;
+    const mainElement = document.querySelector('main')!
+    const padding = getPadding(mainElement)
+    // const allowedContentFullWidth = screenWidth - padding
+    const allowedContentFullWidth = screenWidth
+
+    const contentWidth = getPseudoElementWidth(currentElement, ':before');
+    const elementScrrenXPos = currentElement.getBoundingClientRect().x
+    const elementStaticXPos = currentElement.offsetLeft
+    const elementScrrenXPosRatio = elementScrrenXPos / allowedContentFullWidth
+
+    return {
+        elementStaticXPos,
+        elementScrrenXPos,
+        contentWidth,
+        allowedContentFullWidth,
+        elementScrrenXPosRatio,
+        padding
+    }
+}
+
+export function useCustomStyle() {
```

Comment:
This seems like a ton of work just to always position a tooltip a static amount of pixels from the left of it's relative container in mobile.. The same-or-similar behavior can be achieved by a simpler mechanism. Something similar to the below maybe?
```ts
const styles = getComputedStyle(hint, ':before')
const width = parseInt(styles.width.replace('px',''))
const left = parseInt(styles.left.replace('px',''))
const maxWidth = codeContainer.getBoundingRect().width
const moveLeftBy = Math.min(0, maxWidth - width - left)
```
You can reach out in discord if you want to talk more about it

Reply:
you are right. this is better